pymc/tests/test_sampling.py

```diff
@@ -1446,3 +1449,171 @@ def test_no_init_nuts_compound(caplog):
         b = pm.Poisson("b", 1)
         pm.sample(10, tune=10)
         assert "Initializing NUTS" not in caplog.text
+
+
+class TestCompileForwardSampler:
```

Comment:
We don't seem to have any test for when an unobserved RV depends on MutableData. What is supposed to happen in that case?

```python
with pm.Model() as m:
  prior_mean = pm.MutableData("prior_means", np.arange(5))
  x = pm.Normal("x", mu=prior_means)
  y = pm.Normal("y", mu=x, observed=np.arange(5))
```

Reply:
The `x` random variable will have a volatile parameter, and will always be interpreted as a volatile variable. That case is covered by `test_volatile_parameters` where we make a parameter volatile or not depending on whether it's included in the trace or not.